src/effects/effectbuttonparameterslot.cpp

```diff
@@ -113,33 +119,39 @@ void EffectButtonParameterSlot::slotValueChanged(double v) {
 
 QDomElement EffectButtonParameterSlot::toXml(QDomDocument* doc) const {
     QDomElement parameterElement;
-    if (m_pEffectParameter != nullptr) {
-        parameterElement = doc->createElement(EffectXml::Parameter);
-        XmlParse::addElement(*doc, parameterElement,
-                             EffectXml::ParameterValue,
-                             QString::number(m_pControlValue->get()));
-    }
+    // if (m_pEffectParameter != nullptr) {
+    //     parameterElement = doc->createElement(EffectXml::Parameter);
+    //     XmlParse::addElement(*doc, parameterElement,
+    //                          EffectXml::ParameterValue,
+    //                          QString::number(m_pControlValue->get()));
+    // }
 
     return parameterElement;
 }
 
 void EffectButtonParameterSlot::loadParameterSlotFromXml(const QDomElement&
                                                   parameterElement) {
-    if (m_pEffectParameter == nullptr) {
-        return;
-    }
-    if (!parameterElement.hasChildNodes()) {
-        m_pControlValue->reset();
-    } else {
-        bool conversionWorked = false;
-        double value = XmlParse::selectNodeDouble(parameterElement,
-                                                  EffectXml::ParameterValue,
-                                                  &conversionWorked);
-        if (conversionWorked) {
-            // Need to use setParameterFrom(..., nullptr) here to
-            // trigger valueChanged() signal emission and execute slotValueChanged()
-            m_pControlValue->setParameterFrom(value, nullptr);
-        }
-        // If the conversion failed, the default value is kept.
-    }
+    // if (m_pEffectParameter == nullptr) {
+    //     return;
+    // }
+    // if (!parameterElement.hasChildNodes()) {
+    //     m_pControlValue->reset();
+    // } else {
+    //     bool conversionWorked = false;
+    //     double value = XmlParse::selectNodeDouble(parameterElement,
+    //                                               EffectXml::ParameterValue,
+    //                                               &conversionWorked);
+    //     if (conversionWorked) {
+    //         // Need to use setParameterFrom(..., nullptr) here to
+    //         // trigger valueChanged() signal emission and execute slotValueChanged()
+    //         m_pControlValue->setParameterFrom(value, nullptr);
+    //     }
+    //     // If the conversion failed, the default value is kept.
+    // }
+}
+
+void EffectButtonParameterSlot::syncSofttakeover() {
```

Comment:
Instead of having somewhat confusing blank functions in this class for `syncSofttakeover` and `onEffectMetaParameterChanged`, I think it would be better to have the default implementations of these functions in EffectParameterSlotBase be empty functions and override the default implementation in EffectKnobParameterSlot. That way, EffectButtonParameterSlot wouldn't need any code relating to these functions.